boxsdk/object/item.py

```diff
@@ -613,3 +621,94 @@ def get_collaborations(self, limit=None, marker=None, fields=None):
+        Applies metadata classification for the specified :class:`File` or :class:`Folder` object.
+
+        :param classification:
+            The classification to add to the :class:`File` or :class:`Folder`
+        :type classification:
+            `unicode`
+        :return:
+            The classification added to the :class:`File` or :class:`Folder.
+        :rtype:
+            `unicode`
+        """
+        classification_metadata = {
+            'Box__Security__Classification__Key': classification,
+        }
+        template_key = 'securityClassification-6VMVochwUWo'
+        metadata_classification = self.metadata('enterprise', template_key).create(classification_metadata)
+        return metadata_classification['Box__Security__Classification__Key']
+
+    def update_classification(self, classification):
+        """
+        Updates metadata classification for the specified :class:`File` or :class:`Folder` object.
+
+        :param classification:
+            The classification to add to the :class:`File` or :class:`Folder`
+        :type classification:
+            `unicode`
+        :return:
+            The classification updated on the :class:`File` or :class:`Folder.
+        :rtype:
+            `unicode`
+        """
+        classification_metadata = self.metadata('enterprise', 'securityClassification-6VMVochwUWo')
+        updates = classification_metadata.start_update()
+        updates.update('/Box__Security__Classification__Key', classification)
+        metadata_classification = classification_metadata.update(updates)
+        return metadata_classification['Box__Security__Classification__Key']
+
+    def set_classification(self, classification):
+        """
+        Attempts to add a metadata classification to a :class:`File` or :class:`Folder`, if classification exists, then
+        do update.
+
+        :param classification:
+            The classification to add to the :class:`File` or :class:`Folder`
+        :type classification:
+            `unicode`
+        :return:
+            The classification set on the :class:`File` or :class:`Folder.
+        :rtype:
+            `unicode`
+        """
+        try:
+            metadata_classification = self.add_classification(classification)
+        except BoxAPIException as err:
+            if err.status == 409:
+                metadata_classification = self.update_classification(classification)
+            else:
+                raise
+        return metadata_classification
+
+    def get_classification(self):
+        """
+        Retrieves the classification specified for the :class:`File` or :class:`Folder`
+
+        :return:
+            The classification on the :class:`File` or :class:`Folder.
+        :rtype:
+            `unicode`
+        """
+        try:
+            classification = self.metadata('enterprise', 'securityClassification-6VMVochwUWo').get()
+        except BoxAPIException as err:
+            if err.status == 404:
```

Comment:
What happens if the user does something like this?

```python
file = client.file('id_that_does_not_exist').get_classification()
```

I think the 404 could apply to either the file itself or the classification — we would probably want the former case to still throw, but the latter to return `None`

Reply:
@mattwiller in this case are you referring to:

```python
if err.status == 404
	raise
else:
	return None
```

or are you referring to:

1.) Checking if the file.get() is a 404
2.) Throw if file.get() returns 404
3.) Return None if file.get().get_classification() returns 404.

If latter, what is a good way to check for file.get() is 404 outside of making an explicit call to check if file exists first?